pyrefly/lib/commands/check.rs

```diff
@@ -384,9 +387,193 @@ impl OutputFormat {
             Self::MinText => Self::write_error_text_to_console(relative_to, errors, false),
             Self::FullText => Self::write_error_text_to_console(relative_to, errors, true),
             Self::Json => Self::write_error_json_to_console(relative_to, errors),
+            Self::Github => Self::write_error_github_to_console(relative_to, errors),
             Self::OmitErrors => Ok(()),
         }
     }
+
+    fn write_error_github(
+        writer: &mut impl Write,
+        relative_to: &Path,
+        errors: &[Error],
+    ) -> anyhow::Result<()> {
+        for error in errors {
+            if let Some(command) = github_actions_command(error, relative_to) {
+                writeln!(writer, "{command}")?;
+            }
+        }
+        Ok(())
+    }
+
+    fn buffered_write_error_github(
+        writer: impl Write,
+        relative_to: &Path,
+        errors: &[Error],
+    ) -> anyhow::Result<()> {
+        let mut writer = BufWriter::new(writer);
+        Self::write_error_github(&mut writer, relative_to, errors)?;
+        writer.flush()?;
+        Ok(())
+    }
+
+    fn write_error_github_to_file(
+        path: &Path,
+        relative_to: &Path,
+        errors: &[Error],
+    ) -> anyhow::Result<()> {
+        let file = File::create(path)?;
+        Self::buffered_write_error_github(file, relative_to, errors)
+    }
+
+    fn write_error_github_to_console(relative_to: &Path, errors: &[Error]) -> anyhow::Result<()> {
+        Self::buffered_write_error_github(stdout(), relative_to, errors)
+    }
+}
+
+fn should_emit_github_annotations() -> bool {
+    matches!(
+        std::env::var("GITHUB_ACTIONS"),
```

Comment:
Is this a requirement? Or would passing in `--output=github` be sufficient? I would prefer the CLI command, but want to understand if there's a reason for this.

Reply:
Because GitHub ci will set this env var automatically. So by this, it can detect itself is in ci, and print diagnosis.